src/Components/Facility/ConsultationDetails/index.tsx

```diff
@@ -120,16 +120,21 @@ export const ConsultationDetails = (props: any) => {
             data.symptoms_text = symptoms.join(", ");
           }
           setConsultationData(data);
-          const assetRes = await dispatch(
-            listAssetBeds({
-              bed: data?.current_bed?.bed_object?.id,
-            })
-          );
-          const isCameraAttachedRes = assetRes.data.results.some(
-            (asset: { asset_object: { asset_class: string } }) => {
-              return asset?.asset_object?.asset_class === "ONVIF";
-            }
-          );
+          const assetRes = data?.current_bed?.bed_object?.id
+            ? await dispatch(
+                listAssetBeds({
+                  bed: data?.current_bed?.bed_object?.id,
+                })
+              )
+            : null;
+          const isCameraAttachedRes =
+            assetRes != null
+              ? assetRes.data.results.some(
+                  (asset: { asset_object: { asset_class: string } }) => {
```

Comment:
Avoid redefining types. It would end up being hard to maintain later on.

```suggestion
                  (asset: AssetBedModel) => {
```

Reply:
Updated